include/godot_cpp/variant/aabb.hpp

```diff
@@ -336,7 +336,7 @@ inline void AABB::expand_to(const Vector3 &p_vector) {
 }
 
 void AABB::project_range_in_plane(const Plane &p_plane, real_t &r_min, real_t &r_max) const {
-	Vector3 half_extents(size.x * 0.5, size.y * 0.5, size.z * 0.5);
+	Vector3 half_extents(size.x * 0.5f, size.y * 0.5f, size.z * 0.5f);
```

Comment:
This might break with `REAL_T_IS_DOUBLE`, the fix might be to cast the other one to `real_t` instead. CC @aaronfranke

Reply:
wouldn't it upcast say `0.5f` to double as it's multiplied with another double, instead of downcast the double?

Reply:
Powers of two can be represented exactly with floats, so `0.5f == 0.5`, so it works fine here.

Other options include `* (real_t)0.5` or `/ 2` but they should all get compiled to the same instructions, so it's a matter of style. The only one with different behavior is `* 0.5` because it casts up to double.

Reply:
I think we need to make a choice for consistency. Either we use `(real_t)0.5` everywhere or we use `0.5f`. See the conversation down below about `(real_t)1.0` v.s. `1.0f`.